### PR TITLE
[Fix] Pads query_start_loc to satisfy FIA/TND constraint

### PR DESCRIPTION
**Pull request by yiz-liu**
### What this PR does / why we need it?
This handles both uniform and mixed batches (by inserting a dummy request for mixed batches), consolidates ad-hoc padding into a single helper, copies the updated buffer to the device, and asserts the layout constraint before building the attention metadata. Together, these changes prevent kernel mismatches or failures and ensure correct shapes for FIA/TND execution in full graph modes.

We currently place this helper in `execute_model`. My original design was to include it in `_prepare_inputs`, but that doesn’t work because it must run after padding. While I’d prefer to minimize the impact and reuse as much of the base class as possible in the future, it doesn’t seem achievable at the moment.

### Does this PR introduce _any_ user-facing change?
None.

### How was this patch tested?
Test cases added.

- vLLM version: v0.14.1
- vLLM main: https://github.com/vllm-project/vllm/commit/dc917cceb877dfd13f98c538c4c96158047d98bd

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -509,6 +516,36 @@ def get_model(self) -> nn.Module:
...
+        if num_tokens_padded == num_reqs_padded * self.uniform_decode_query_len:
+            # Uniform-batch case: num_reqs must be no greater than num_reqs_padded
+            assert num_reqs <= num_reqs_padded
+
+            last_loc = self.query_start_loc.np[num_reqs]
+            self.query_start_loc.np[num_reqs + 1 : num_reqs_padded + 1] = (
+                self.arange_np[1 : num_reqs_padded + 1 - num_reqs]
+                * self.uniform_decode_query_len
+                + last_loc
+            )
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py:536]: [0, 1, 2] -> [0, 1, 2, 3, 4]

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -509,6 +516,36 @@ def get_model(self) -> nn.Module:
...
+        else:
+            # Mixed-batch case: num_reqs must equal num_reqs_padded
+            assert num_reqs == num_reqs_padded
+
+            # Insert a dummy request instead of setting query_start_loc[num_reqs] = num_tokens_padded directly
+            self.query_start_loc.np[num_reqs_padded + 1] = num_tokens_padded
+            num_reqs_padded = num_reqs_padded + 1
+
+        self.query_start_loc.copy_to_gpu()
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py:545]: [0, 3] -> [0, 3, 4]

File: vllm_ascend/worker/model_runner_v1.py [hunk 3]
```diff
@@ -206,6 +206,13 @@ def __init__(self, vllm_config: VllmConfig, device: torch.device):
         vllm_config.scheduler_config.max_num_batched_tokens += max_pcp_pad_tokens
         with _torch_cuda_wrapper():
             super().__init__(vllm_config, device)
+
+        # NOTE: For FULL mode we change +1 to +2 to reserve extra space for padding.
+        # See _pad_query_start_loc_for_fia.
+        self.query_start_loc = self._make_buffer(
+            self.max_num_reqs + 2, dtype=torch.int32  # type: ignore[has-type]
+        )
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py:214]: Check if other buffers should be extended too.

File: vllm_ascend/worker/model_runner_v1.py [hunk 4]
```diff
@@ -1207,6 +1241,11 @@ def execute_model(
                 use_spec_decode = len(scheduler_output.scheduled_spec_decode_tokens) > 0
                 ubatch_slices_attn = ubatch_slices_padded if pad_attn else ubatch_slices
 
+                if cudagraph_mode != CUDAGraphMode.NONE:
+                    num_reqs_padded = self._pad_query_start_loc_for_fia(
+                        num_tokens_padded, num_reqs_padded, num_reqs
+                    )
+
```
Comment (yiz-liu) [inline vllm_ascend/worker/model_runner_v1.py:1248]: Maybe current_platform.post_process_after_padding

File: vllm_ascend/worker/model_runner_v1.py (hunk 3 above)
Comment (yiz-liu) [reply vllm_ascend/worker/model_runner_v1.py:214]: This strange, no error by now.
